shared-module/displayio/Group.c

```diff
@@ -34,6 +34,11 @@
 #include "shared-bindings/vectorio/VectorShape.h"
 #endif
 
+static void check_rom(displayio_group_t *self) {
+    if (self->in_rom) {
```

Comment:
Instead of adding a flag, how about testing that `members` is on the VM heap with `gc_nbytes()`? That way we don't need to make sure the flag is set. The group itself isn't const so it is in RAM. It can be mutated. Only the list won't work well with it in static RAM vs the VM heap.

Reply:
I tried this suggestion, but it was worse for code size. Instead, I renamed `check_rom` and `in_rom` (I'll push that in a moment)

I think this is because checking the bit is just an instruction or 2, while checking the gc size is some pointer dereferencing and a function call. Still, I was surprised by the amount of code growth, about 22 bytes per check_readonly call (there are 7), so I may have done something wrong. The `check_readonly` function was _inlined_ on the pico w build so that makes the degree of growth a little more expected.

Something else interesting to note about the bitfield approach: it doesn't actually require any more storage (there was a bit available) or any more code in the constructor (the compiler can clear multiple bitfields in the same word efficiently) so I think it's probably the best approach for code size even if using `gc_nbytes` is conceptually simpler.

<details>
<summary>
The change I tried, which grew the binary by 156 bytes (pico w build):
</summary>

```diff
commit da8364208863fbc546d1edf510d1f4b3dc587e88
Author: Jeff Epler <jepler@gmail.com>
Date:   Thu Feb 15 21:59:36 2024 -0600

    use gc_nbytes instead of flag to check for readonly

diff --git a/shared-module/displayio/Group.c b/shared-module/displayio/Group.c
index d20ad5ebe6..90f91c52c3 100644
--- a/shared-module/displayio/Group.c
+++ b/shared-module/displayio/Group.c
@@ -26,6 +26,7 @@
 
 #include "shared-bindings/displayio/Group.h"
 
+#include "py/gc.h"
 #include "py/runtime.h"
 #include "py/objlist.h"
 #include "shared-bindings/displayio/TileGrid.h"
@@ -35,7 +36,7 @@
 #endif
 
 static void check_readonly(displayio_group_t *self) {
-    if (self->readonly) {
+    if (!gc_nbytes(self->members->items)) {
         mp_raise_RuntimeError(MP_ERROR_TEXT("Read-only"));
     }
 }
@@ -383,7 +384,6 @@ void displayio_group_construct(displayio_group_t *self, mp_obj_list_t *members,
     self->item_removed = false;
     self->scale = scale;
     self->in_group = false;
-    self->readonly = false;
 }
 
 bool displayio_group_fill_area(displayio_group_t *self, const _displayio_colorspace_t *colorspace, const displayio_area_t *area, uint32_t *mask, uint32_t *buffer) {
diff --git a/shared-module/displayio/Group.h b/shared-module/displayio/Group.h
index 6a9432adfb..f684223154 100644
--- a/shared-module/displayio/Group.h
+++ b/shared-module/displayio/Group.h
@@ -47,8 +47,7 @@ typedef struct {
     bool in_group : 1;
     bool hidden : 1;
     bool hidden_by_parent : 1;
-    bool readonly : 1;
-    uint8_t padding : 3;
+    uint8_t padding : 4;
 } displayio_group_t;
 
 void displayio_group_construct(displayio_group_t *self, mp_obj_list_t *members, uint32_t scale, mp_int_t x, mp_int_t y);
diff --git a/supervisor/shared/display.c b/supervisor/shared/display.c
index 846d7e3fb0..f7061083a1 100644
--- a/supervisor/shared/display.c
+++ b/supervisor/shared/display.c
@@ -208,5 +208,4 @@ displayio_group_t circuitpython_splash = {
     .in_group = false,
     .hidden = false,
     .hidden_by_parent = false,
-    .readonly = true,
 };
```
</details>